client/py/synnax/simulation/auto.py

```diff
@@ -0,0 +1,153 @@
+
+    def step(self, state: dict[str, float]) -> dict[str, float]:
+        self.step_num += 1
+
+        if state[self.pt] >= 400:
+            state[self.press] = 0
+        else:
+            state[self.press] = 1
+
+        if state[self.press] == 1 and state[self.pre] == 1:
+            state[self.pt] += 1
+        elif state[self.press] == 0 and state[self.pre] == 0:
+            state[self.pt] -= 1
+        else:
+            state[self.pt] += 0
+
+        return state
+
+
+class Nominal:
+    def __init__(self, state_machine, error_step: int):
+        self.state_machine = state_machine
+        self.error_step = error_step
+
+    def step(self, state: dict[str, float], step_num: int):
+        self.state_machine.step(state)
+        return state
+
+
+# Anomaly Implementations
+
+
+class VoltageZero:
+    def __init__(self, state_machine, error_step: int):
+        self.state_machine = state_machine
+        self.error_step = error_step
+
+    def step(self, state: dict[str, float], step_num: int):
+        if step_num < self.error_step:
+            state[self.state_machine.pt] = -200
+        self.state_machine.step_num += 1
+        return state
+
+
+class VoltageFull:
+    def __init__(self, state_machine, error_step: int):
+        self.state_machine = state_machine
+        self.error_step = error_step
+
+    def step(self, state: dict[str, float], step_num: int):
+        if step_num < self.error_step:
+            state[self.state_machine.pt] = 1000
+        self.state_machine.step_num += 1
+        return state
+
+
+class PressValveStuckOpen:
+    def __init__(self, state_machine, error_step: int):
+        self.state_machine = state_machine
+        self.error_step = error_step
+
+    def step(self, state: dict[str, float], step_num: int):
+        if step_num < self.error_step:
+            state[self.state_machine.press] == 1
+            if state[self.state_machine.pre] == 0:  ##open, pressure stays constant
+                state[self.state_machine.pt] += 0
+            else:
+                state[self.state_machine.pt] += 1
+        return self.state_machine.step(state)
+
+
+class PreValveStuckOpen:
+    def __init__(self, state_machine, error_step: int):
+        self.state_machine = state_machine
+        self.error_step = error_step
+
+    def step(self, state: dict[str, float], step_num: int):
+        if step_num < self.error_step:
+            if state[self.state_machine.press] == 1:  ##open, pressure stays constant
+                state[self.state_machine.pt] += 0
+            else:
+                state[self.state_machine.pt] -= 1
+        return self.state_machine.step(state)
+
+
+# Base simulator class
+
+
+class Simulator:
+    frequency: sy.Rate.HZ = 100 * sy.Rate.HZ
+    state: dict[str, float]
+    simulated_responders: list[SimulatedResponse]
+    anomaly: Anomaly
+
+    def __init__(
+        self,
+        input_state,
+        input_responders: list[SimulatedResponse],
+        input_anomaly: Anomaly,
```

Comment:
This is strange. Anomalies are just simulated responses, and should be treated as such. Composition of simulated responses should occur outside of this class.